loader_hub/graphql/base.py

```diff
@@ -0,0 +1,73 @@
+"""GraphQL Reader."""
+
+from typing import Dict, List, Optional
+
+from llama_index.readers.base import BaseReader
+from llama_index.readers.schema.base import Document
+import yaml
+
+
+class GraphQLReader(BaseReader):
+    """GraphQL reader.
+
+    Combines all GraphQL results into the Document used by LlamaIndex.
+
+    Args:
+        uri (str): GraphQL uri.
+        headers (Dict): Optional http headers.
+
+    """
+
+    def __init__(
+        self,
+        uri: Optional[str] = None,
+        headers: Dict = {},
+    ) -> None:
+        """Initialize with parameters."""
+        try:
+            from gql import Client
+            from gql.transport.requests import RequestsHTTPTransport
+
+        except ImportError:
+            raise ImportError(
+                "`gql` package not found, please run `pip install gql`"
+            )
+        if uri:
+            if uri is None:
+                raise ValueError("`uri` must be provided.")
+            transport = RequestsHTTPTransport(url=uri, headers=headers)
+            self.client = Client(transport=transport, fetch_schema_from_transport=True)
+            
+    def load_data(
+        self, query: str, variables: Dict = {}
```

Comment:
nit: I would type as `Optional[Dict]` and set default value to None here, and then in the function call do `variables = variables or {}`. Setting `{}` in the default params can have unintended side effects